src/storage/backends/mongo.ts

```diff
@@ -0,0 +1,90 @@
+import * as env from "env-var";
+import {
+  MongoClient,
+  MongoClientOptions,
+  InsertOneResult,
+  FindCursor,
+  Document,
+} from "mongodb";
+
+export const createMongoClient = (dbName: string): MongoClient => {
+  const user = env.get("MONGO_USERNAME").required().asString();
+  const pass = env.get("MONGO_PASSWORD").required().asString();
+  const uri = env.get("MONGO_CLUSTER_URI").required().asString();
+
+  const fullUri = `mongodb+srv://${user}:${pass}@${uri}/${dbName}`;
+
+  const options: MongoClientOptions = {
+    connectTimeoutMS: 5000,
+    retryWrites: true,
+    w: "majority",
+  };
+
+  const client = new MongoClient(fullUri, options);
+
+  return client;
+};
+
+// Will create the collection if it does not already exist
+export const insertOne = async (
```

Comment:
You need to stop code-smell as soon as possible, this should be generic / templated like such:
```
export const insertOne = async <T>(
  data: T,
  db: string,
  collection: string
): Promise<InsertOneResult<T>> => {
  const client: MongoClient = createMongoClient(db);
  try {
    await client.connect();
    const database = client.db(db);
    const usedCollection = database.collection(collection);

    const result: InsertOneResult<T> = await usedCollection.insertOne(data);

    return result;
  } catch (err) {
    throw Error;
  } finally {
    await client.close();
  }
};
```